transforms/manual-bind-to-arrow.js

```diff
@@ -0,0 +1,85 @@
+/**
+ * Copyright 2013-2015, Facebook, Inc.
+ * All rights reserved.
+ *
+ * This source code is licensed under the BSD-style license found in the
+ * LICENSE file in the root directory of this source tree. An additional grant
+ * of patent rights can be found in the PATENTS file in the same directory.
+ *
+ */
+
+/**
+ * class Component extends React.Component {
+ *   constructor() { this.onClick = this.onClick.bind(this); }
+ *   onClick() { }
+ * }
+ *
+ * -->
+ *
+ * class Component extends React.Component {
+ *   constructor() { }
+ *   onClick = () => { }
+ * }
+ */
+
+export default function transformer(file, api) {
+  const j = api.jscodeshift;
+  const {expression, statement, statements} = j.template;
+
+  var root = j(file.source);
+
+  return root
+    .find(j.AssignmentExpression)
+    .filter(path => {
+      // Check constructor
+      var methodPath = path;
+      while (methodPath &&
+             (methodPath.node.type !== 'MethodDefinition' ||
+              methodPath.node.kind !== 'constructor')) {
+        methodPath = methodPath.parentPath;
+      }
+      if (!methodPath) {
+        return false;
+      }
+
+      if (!(
+        path.node.left.type === 'MemberExpression' &&
```

Comment:
I like having it in one giant if, I can just comment stuff out and console.log things in the process. I'm not sure it would be really more readable or anything splitting them out in variables.